packages/x-date-pickers/src/models/adapters.ts

```diff
@@ -721,9 +721,16 @@ export interface MuiPickersAdapter<TDate, TLocale = any> {
   mergeDateAndTime(dateParam: TDate, timeParam: TDate): TDate;
   /**
    * Get the label of each day of a week.
+   * @deprecated Will be removed in v7. Use `adapter.getWeek` and format the Dates.
    * @returns {string[]} The label of each day of a week.
    */
   getWeekdays(): string[];
+  /**
+   * Get an array with all the weekdays dates in the week of the given date.
+   * @param {TDate} value The given date.
+   * @returns {TDate[]} The date of each day of a week.
+   */
+  getWeek(value: TDate): TDate[];
```

Comment:
Did you encountered anything preventing you from not putting this logic inside the adapter?

See https://github.com/mui/mui-x/issues/9984#issuecomment-1717602090

Reply:
Initially, I was talking about deprecating `getWeekDays` and add a new adapter method that does the same but returning dates.

BUT, I saw that if the new `getWeekDays` returns dates, then it can be created outside of the adapter (see my comment linked above).
So we can just deprecate `adapters.getWeekDays` and create a new utility method in the picker package that returns dates and use it internally.

____

For `dayOfWeekFormatter`, we clearly have a breaking change and for me we need to do it in a non-breaking way :+1: